clients/java/src/test/java/io/camunda/client/tenant/AssignRoleToTenantTest.java

```diff
@@ -0,0 +1,91 @@
+ * limitations under the License.
+ */
+package io.camunda.client.tenant;
+
+import static io.camunda.client.impl.http.HttpClientFactory.REST_API_PATH;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import io.camunda.client.api.command.ProblemException;
+import io.camunda.client.protocol.rest.ProblemDetail;
+import io.camunda.client.util.ClientRestTest;
+import io.camunda.client.util.RestGatewayService;
+import org.junit.jupiter.api.Test;
+
+public class AssignRoleToTenantTest extends ClientRestTest {
+
+  private static final String TENANT_ID = "foo";
+  private static final String ROLE_ID = "roleId";
+
+  @Test
+  void shouldAssignRoleToTenant() {
+    // when
+    client.newAssignRoleToTenantCommand(TENANT_ID).roleId(ROLE_ID).send().join();
+
+    // then
+    final String requestPath = RestGatewayService.getLastRequest().getUrl();
+    assertThat(requestPath)
+        .isEqualTo(REST_API_PATH + "/tenants/" + TENANT_ID + "/roles/" + ROLE_ID);
+  }
+
+  @Test
+  void shouldRaiseExceptionOnNotFoundTenant() {
+    gatewayService.errorOnRequest(
+        REST_API_PATH + "/tenants/" + TENANT_ID + "/roles/" + ROLE_ID,
+        () -> new ProblemDetail().title("Not Found").status(404));
+
+    assertThatThrownBy(
+            () -> client.newAssignRoleToTenantCommand(TENANT_ID).roleId(ROLE_ID).send().join())
+        .isInstanceOf(ProblemException.class)
+        .hasMessageContaining("Failed with code 404: 'Not Found'");
+  }
+
+  @Test
+  void shouldRaiseExceptionOnNotFoundRole() {
+    gatewayService.errorOnRequest(
+        REST_API_PATH + "/tenants/" + TENANT_ID + "/roles/" + ROLE_ID,
+        () -> new ProblemDetail().title("Not Found").status(404));
+
+    assertThatThrownBy(
+            () -> client.newAssignRoleToTenantCommand(TENANT_ID).roleId(ROLE_ID).send().join())
+        .isInstanceOf(ProblemException.class)
+        .hasMessageContaining("Failed with code 404: 'Not Found'");
+  }
+
+  @Test
+  void shouldHandleServerError() {
+    gatewayService.errorOnRequest(
+        REST_API_PATH + "/tenants/" + TENANT_ID + "/roles/" + ROLE_ID,
+        () -> new ProblemDetail().title("Internal Server Error").status(500));
+
+    assertThatThrownBy(
+            () -> client.newAssignRoleToTenantCommand(TENANT_ID).roleId(ROLE_ID).send().join())
+        .isInstanceOf(ProblemException.class)
+        .hasMessageContaining("Failed with code 500: 'Internal Server Error'");
+  }
+
+  @Test
+  void shouldRaiseExceptionOnForbiddenRequest() {
+    gatewayService.errorOnRequest(
+        REST_API_PATH + "/tenants/" + TENANT_ID + "/roles/" + ROLE_ID,
+        () -> new ProblemDetail().title("Forbidden").status(403));
+
+    assertThatThrownBy(
+            () -> client.newAssignRoleToTenantCommand(TENANT_ID).roleId(ROLE_ID).send().join())
+        .isInstanceOf(ProblemException.class)
+        .hasMessageContaining("Failed with code 403: 'Forbidden'");
+  }
```

Comment:
🔧 These tests can go, they don't provide any value as you are testing the mock and the generic exception handling which is already covered elsewhere.

What I'm missing is the validations that the `roleId` and `tenantId` may not be null or empty.